Blog: Eval Villain

psiinon opened the pull request:
TODO

- [x] Add images when received from @swoops
- [x] Confirm author display name and add pic for @swoops (if wanted)
- [ ] Try to colourise the `[EV] Functions hooked for <<url>>` line
- [x] Publish EV add-on

I _think_ I've got all of the _bold_ and *italic* but reviewers with access to the original doc please check :)

![Screenshot 2021-11-30 at 10-43-42 OWASP ZAP – The Eval Villain Add-on](https://user-images.githubusercontent.com/1081115/144033307-64f37986-aa89-4932-8b1c-c4af23ed6dd2.png)


Signed-off-by: Simon Bennetts <psiinon@gmail.com>

kingthorin commented on site/content/blog/2021-11-30-the-eval-villain-add-on/index.md:
```diff
@@ -0,0 +1,105 @@
+
+## Eval Villain Quick Start 
+
+To get started with Eval Villain launch Firefox from ZAP, select the Eval Villain icon on the top right hand side of Firefox, click the enable switch and reload the current web page. Now open the console and you should see a message at the top like "[EV] Functions hooked for <<url>>". This means Eval Villain is loaded and is working for the current URL. If there are DOM interactions you may see m ...
```
Is this referring to the devtools console or something in Eval Villain?

swoops replied:
Dev tools console. That is a good point, readers don't know that Eval Villain does not have a console. So maybe it would be good to make it more explicit. Like "open the developer tools and go to the console there you should see...".

swoops replied:
Dev tools console. That is a good point, readers don't know that Eval Villain does not have a console. So maybe it would be good to make it more explicit. Like "open the developer tools and go to the console there you should see...".

kingthorin commented on site/content/blog/2021-11-30-the-eval-villain-add-on/index.md:
```suggestion
In essence, Eval Villain is building the user configuration into a JavaScript string and forcing it into a web page before the web page has had a chance to run its own JavaScript. Since there is no race with the web page, Eval Villain will even hook functions before inline JavaScript has a chance to run on a local "file://" scheme.
```

swoops replied:
That change is correct, nice catch. I can't don't think I can update the pull or I would approve it or something.

kingthorin replied:
Poke @psiinon :wink:

psiinon commented:
Think I've fixed all of those comments.
Have added the images (thanks @swoops ) and changed the publish date to tomorrow to give everyone time to review it.

![Uploading Screenshot 2021-12-01 at 12-23-08 OWASP ZAP – The Eval Villain Add-on.png…]()

thc202 commented on site/content/blog/2021-12-02-the-eval-villain-add-on/index.md:
```diff
@@ -0,0 +1,105 @@
+
+![Eval Villain start](images/ev_first.png)
+
+Eval Villain notified you that the hooked function `eval` was used. You can see the argument passed to eval was ‘`alert("Welcome to the website!")`, and a stack trace shows where the function was called. The top two functions in the stack trace are actually a part of Eval Villain—but they are left in to make everything transparent and debuggable.
```
Remove `‘`.

thc202 commented on site/content/blog/2021-12-02-the-eval-villain-add-on/index.md:
`eye on localStorage` → `eye on local storage`

thc202 commented:
The date could be changed to today.

thc202 commented:
Thank you!
